app/services/usps_in_person_proofing/date_validator.rb

```diff
@@ -0,0 +1,37 @@
+module UspsInPersonProofing
+  # Validator that can be attached to a form or other model
+  # to verify that specific supported fields are comparable to other dates.
+  # Since it's a subclass of ComparisonValidator, it share the same options.
+  # == Example
+  #
+  #   validates_with UspsInPersonProofing::DateValidator,
+  #     attributes: [:dob],
+  #     message: "error",
+  #     less_than_or_equal_to: ->(_rec) {
+  #         Time.zone.today - IdentityConfig.store.idv_min_age_years.years
+  #     }
+  #     ....
+  #
+  class DateValidator < ActiveModel::Validations::ComparisonValidator
+    private
+
+    def prepare_value_for_validation(value, _record, _attr_name)
+      val_to_date(value)
+    rescue
+      nil
+    end
+
+    # @param [String,Date,#to_hash] param
+    # @return [Date]
+    # It's caller's responsibility to ensure the param contains required entries
```

Comment:
i'm not quite sure I understand this. what are required entries?

Reply:
@svalexander, the param can be a type of ActionController::Parameters from state id form submission, where it contains 3 keys, year, month, day. Here we treat it as a hash, so if you pass in anything can be treated as a hash(.to_hash method), it assumes that these three entries exists, it's caller's responsibility to make sure they exists to avoid error.

Reply:
ahh ok thank you for clarifying